.github/workflows/pull_request.yml

```diff
@@ -1,18 +1,39 @@
+name: Build Chain
+
 on: [ workflow_dispatch ]
 
 jobs:
-  build-chain:
+  build-chain-openjdk8:
+    runs-on: ubuntu-latest
+    name: Pull Request openjdk8
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build Chain
+      id: build-chain
+      uses: kiegroup/github-action-build-chain@openjdk8
+      with:
+        parent-dependencies: 'drools@7.x:master'
+        child-dependencies: 'jbpm@7.x:master'
+        build-command: 'mvn -e clean install -nsu -Prun-code-coverage,wildfly -Dfull  -Dintegration-tests=true -Dmaven.test.failure.ignore=true -DjvmArgs="-Xms1g -Xmx4g" | mvn -nsu generate-resources -Psonarcloud-analysis -DjvmArgs="-Xms1g -Xmx4g"'
+        build-command-upstream: "mvn -e clean install -DskipTests -Dgwt.compiler.skip=true -Dgwt.skipCompilation=true -Denforcer.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true -Drevapi.skip=true"
+        workflow-file-name: "pull_request.yml"
+      env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+  build-chain-openjdk11:
     runs-on: ubuntu-latest
-    name: Build Pull Request
+    name: Pull Request openjdk11
     steps:
     - uses: actions/checkout@v2
     - name: Build Chain
       id: build-chain
-      uses: kiegroup/github-action-build-chain@master
+      uses: kiegroup/github-action-build-chain@openjdk11
```

Comment:
It is unfortunate that you need to specify JDK like this. If you could use the standard way (see https://github.com/kiegroup/optaplanner/blob/master/.github/workflows/os_java_ci.yml#L29), you'd be able to use a standard matrix build, avoiding this large config duplication.

Reply:
I think @Ginxo wrote about some limitation on GA side there is some bug which blocks this easier way of config

from @Ginxo`s msg:
> just to let you know I tried to use matrix in order not to duplicate commands for openjdk8 and 11, but it's not possible to use it in usues like uses: kiegroup/github-action-build-chain@{{ matrix.build-chain-version}}
the limitation is already reported and there's no way of work around it (apart from duplicate it again)
The other thing you should pay attention to is workflow-file-name since the name of the flow is defined there's no way about how to get file-name information (I already reported it to github) (edited)

Reply:
@triceo that's it, the thing is build-chain tool runs over a docker image (java8+maven3.6.6, java11+maven3.6.3, gradle, whatever) but there's not way to dynamically select the build-chain version you want to use. So for now the only way is to duplicate it.

Reply:
https://github.community/t/using-strategy-matrix-values-in-the-specification-of-actions-to-be-used-in-a-workflow/16706